auto3dseg/algorithm_templates/segresnet/scripts/validate.py

```diff
@@ -9,204 +9,20 @@
-
-            with torch.cuda.amp.autocast():
-                d["pred"] = sliding_window_inference(
-                    val_images, patch_size_valid, num_sw_batch_size, model, mode="gaussian", overlap=overlap_ratio
-                )
-
-            d = [post_transforms(i) for i in decollate_batch(d)]
-            val_outputs = post_pred(d[0]["pred"])
-            val_outputs = val_outputs[None, ...]
-
-            if softmax:
-                val_labels = post_pred(val_labels[0, ...])
-                val_labels = val_labels[None, ...]
-            print(val_images.shape, val_labels.shape, val_outputs.shape, patch_size_valid)
-            value = compute_meandice(y_pred=val_outputs, y=val_labels, include_background=not softmax)
-
-            metric_count += len(value)
-            metric_sum += value.sum().item()
-            metric_vals = value.cpu().numpy()
-            if len(metric_mat) == 0:
-                metric_mat = metric_vals
-            else:
-                metric_mat = np.concatenate((metric_mat, metric_vals), axis=0)
-
-            print_message = ""
-            print_message += str(_index + 1)
-            print_message += ", "
-            print_message += d[0]["pred"].meta["filename_or_obj"]
-            print_message += ", "
-            for _k in range(metric_dim):
-                if output_classes == 2:
-                    print_message += f"{metric_vals.squeeze():.5f}"
-                else:
-                    print_message += f"{metric_vals.squeeze()[_k]:.5f}"
-                print_message += ", "
-            print(print_message)
-
-            row = [d[0]["pred"].meta["filename_or_obj"]]
-            for _i in range(metric_dim):
-                row.append(metric_vals[0, _i])
-
-            with open(os.path.join(output_path, "raw.csv"), "a", encoding="UTF8") as f:
-                writer = csv.writer(f)
-                writer.writerow(row)
-
-            for _c in range(metric_dim):
-                val0 = torch.nan_to_num(value[0, _c], nan=0.0)
-                val1 = 1.0 - torch.isnan(value[0, 0]).float()
-                metric[2 * _c] += val0 * val1
-                metric[2 * _c + 1] += val1
-
-            _index += 1
-
-        metric = metric.tolist()
-        for _c in range(metric_dim):
-            print(f"evaluation metric - class {_c + 1:d}:", metric[2 * _c] / metric[2 * _c + 1])
-        avg_metric = 0
-        for _c in range(metric_dim):
-            avg_metric += metric[2 * _c] / metric[2 * _c + 1]
-        avg_metric = avg_metric / float(metric_dim)
-        print("avg_metric", avg_metric)
-
-        dict_file = {}
-        dict_file["acc"] = float(avg_metric)
-        for _c in range(metric_dim):
-            dict_file["acc_class" + str(_c + 1)] = metric[2 * _c] / metric[2 * _c + 1]
-
-        with open(os.path.join(output_path, "summary.yaml"), "w") as out_file:
-            yaml.dump(dict_file, stream=out_file)
-
-    return
+    override["validate#enabled"] = True
```

Comment:
If the users set `validate#enabled` to False with `override` but we need to override their `override`, can we make an info-level or debug-level log?

Reply:
the default is always False, if we override, it becomes a new rule (it's the same for any other parameter). why do we need to log anything.   PS: may be it's not clear what it means, when it's true it will run 1 validation epoch over the validation set (no training or anything else will be performed)

Reply:
Okay I see what you are saying. I assume the users won't likely do something like:
```
python validate.py --validate#enabled=False
```